### PR TITLE
Make it easier to debug flaky system tests

### PR DESCRIPTION
**Pull request by p-j-smith**
## Description
Toward #465 

This doesn't fix the tests, but makes them easier to debug and gives us more info when the tests do fail. I couldn't reproduce the failing test locally so haven't applied the fix until we understand what the issue is.

- use `pytest-check` to ensure we always test both zip files in `test_ftps_dicom_upload` (previously the test will stop as soon as the first assert failed)
- log which zip file and procedure we are testing in `test_ftps_dicom_upload` 
- add a fixture to cleanup the pixl database after running system tests. This allows system tests to be rerun locally. Previously, when rerunning the system tests, no data would be exported as was already exported on the previous run

## Type of change
Please delete options accordingly to the description.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Suggested Checklist

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have passed on my local host device. (see further details at the [CONTRIBUTING](https://github.com/UCLH-Foundry/PIXL/blob/main/CONTRIBUTING.md#local-testing) document)
- [x] Make sure your branch is up-to-date with main branch. See [CONTRIBUTING](https://github.com/UCLH-Foundry/PIXL/blob/main/CONTRIBUTING.md) for a general example to syncronise your branch with the `main` branch.
- [x] I have requested PR review to `UCLH-Foundtry/arc-dev`
- [ ] I have adddressed and marked as resolved all the review comments in my PR.
- [ ] Finally, I have selected `squash and merge`

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -354,7 +354,7 @@ services:
         ports:
             - "${POSTGRES_PORT}:5432"
         healthcheck:
-            test: ["CMD", "pg_isready", "-U", "${PIXL_DB_USER}"]
+            test: ["CMD", "pg_isready", "-U", "${PIXL_DB_USER}", "--dbname", "${PIXL_DB_NAME}"]
```
Comment (p-j-smith) [inline docker-compose.yml:357]: this is unrelated, but without adding `--dbname` there were confusing errors in the logs from the healthcheck:
> postgres-1  | 2024-08-23 15:08:32.833 GMT [77] FATAL:  database "pixl_db_username" does not exist
postgres-1  | 2024-08-23 15:08:42.882 GMT [85] FATAL:  database "pixl_db_username" does not exist
postgres-1  | 2024-08-23 15:08:52.951 GMT [93] FATAL:  database "pixl_db_username" does not exist

Comment (stefpiatek) [reply docker-compose.yml:357]: ooh yeah that's definitely not what you want

File: test/conftest.py [hunk 2]
```diff
@@ -197,3 +201,28 @@ def _export_patient_data(_setup_pixl_cli) -> None:  # type: ignore [no-untyped-d
     is synchronous (whether that is itself wise is another matter).
     """
     run_subprocess(["pixl", "export-patient-data", str(RESOURCES_OMOP_DIR.absolute())], TEST_DIR)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def _cleanup_database() -> Generator:
```
Comment (stefpiatek) [inline test/conftest.py:207]: 👍🏻

**Review by stefpiatek — APPROVED** [review]
Good call, let's see if we get the error again later on. I'll put the issue for this in the freezer and then if we come across it again we'll be able to debug

**Pull request merged**